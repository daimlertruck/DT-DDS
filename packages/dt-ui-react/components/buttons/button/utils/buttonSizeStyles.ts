import { CustomTheme as Theme } from '../../../../types';
import { ButtonSizes } from '../constants/types';

export const buttonSizeStyles = (size: ButtonSizes, theme: Theme) =>
  ({
    large: `
      ${theme.fontStyles.button1}
      padding: ${theme.spacing['2xs']} ${theme.spacing.s};
    `,
    medium: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['3xs']} ${theme.spacing.xs};
    `,
    small: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['4xs']} ${theme.spacing['2xs']};
    `,
  }[size]);
