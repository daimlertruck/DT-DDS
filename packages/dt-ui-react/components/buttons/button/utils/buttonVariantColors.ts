import { CustomTheme as Theme } from '../../../../types';
import {
  ButtonVariant,
  ContainedColors,
  OutlinedColors,
  TextColors,
} from '../constants/types';

interface ColorKeys {
  color: string;
  bgColor: string;
  hoverBgColor?: string;
  hoverColor?: string;
}

export const buttonVariantColors = (
  color: ContainedColors | OutlinedColors,
  theme: Theme
): Record<ButtonVariant, ColorKeys> => ({
  contained: {
    color: theme.colors.alpha.white,
    bgColor: theme.palette.alpha[color],
    hoverBgColor: theme.palette.alpha[`${color}Dark_300` as ContainedColors],
  },
  outlined: {
    color:
      color === 'neutral'
        ? theme.colors.alpha.gray_200
        : theme.palette.alpha[color],
    bgColor: theme.colors.alpha.white,
    hoverBgColor: theme.palette.alpha[`${color}Light_300` as OutlinedColors],
  },
  text: {
    color: theme.palette.alpha[color],
    bgColor: 'transparent',
    hoverColor: theme.palette.alpha[`${color as TextColors}Dark_300`],
  },
});
