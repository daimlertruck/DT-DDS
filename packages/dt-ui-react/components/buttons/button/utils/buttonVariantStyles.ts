import { CustomTheme as Theme } from '../../../../types';
import { ButtonVariant, OutlinedColors, SolidColors } from '../constants/types';

export const buttonVariantStyles = (
  variant: ButtonVariant,
  theme: Theme,
  color: SolidColors | OutlinedColors
) =>
  ({
    solid: `
      text-transform: uppercase;
      border: ${theme.spacing.none};

      &:focus {
        box-shadow: 0px 2px 1px 0px #1A1A1A0D, 0px 0px 0px 4px ${theme.colors.blue_10};
        background-color: ${theme.palette[color]};
      }

      &:disabled {
        opacity: 38%;
      }
  `,
    outlined: `
      border: 1px solid;
      text-transform: uppercase;
        ${
          color === 'neutral'
            ? theme.palette.neutralDark_100
            : theme.palette[color]
        };

      &:disabled {
        color: ${
          color === 'neutral'
            ? theme.palette.neutralLight_100
            : theme.palette[color]
        };
        background-color: ${theme.colors.white};
        border-color: ${
          color === 'neutral'
            ? theme.palette.neutralLight_100
            : theme.palette[color]
        };
        cursor: default;
      }
    `,
    text: `
      border: none;

      &:disabled {
        color: ${theme.palette.neutralLight_100};
      }
    `,
  }[variant]);
