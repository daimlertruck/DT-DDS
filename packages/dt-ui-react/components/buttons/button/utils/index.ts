export * from './buttonVariantColors';
export * from './buttonVariantStyles';
