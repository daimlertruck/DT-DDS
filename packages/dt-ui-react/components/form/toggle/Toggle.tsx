import { ChangeEvent } from 'react';

import { BaseProps, AlphaColors } from '../../../types';
import { Typography } from '../../typography';

import {
  ToggleCheckBoxStyled,
  ToggleSwitchStyled,
  ToggleWrapperStyled,
} from './Toggle.styled';

export interface ToggleProps extends BaseProps {
  onChange: (evt: ChangeEvent<HTMLInputElement>) => void;
  dataTestId?: string;
  isDisabled?: boolean;
  isChecked?: boolean;
  checkedColor?: AlphaColors;
  uncheckedColor?: AlphaColors;
}

export const Toggle = ({
  onChange,
  dataTestId,
  children,
  checkedColor = 'success',
  uncheckedColor = 'neutral',
  isDisabled = false,
  isChecked = false,
}: ToggleProps) => {
  const testId = dataTestId ?? 'toggle';
  const checkboxId = `${testId}-checkbox-id`;

  return (
    <ToggleWrapperStyled data-testid={testId}>
      <ToggleCheckBoxStyled
        disabled={isDisabled}
        id={checkboxId}
        type='checkbox'
        checkedColor={checkedColor}
        uncheckedColor={uncheckedColor}
        checked={isChecked}
        onChange={onChange}
      />
      <ToggleSwitchStyled htmlFor={checkboxId} />
      {children}
    </ToggleWrapperStyled>
  );
};

Toggle.Label = ({ children, dataTestId }: BaseProps) => {
  return (
    <Typography
      data-testid={dataTestId ?? 'toggle-label'}
      element='span'
      fontStyles='pXXSmall'
      color='gray_200'
      style={{ fontWeight: 700 }}
      dataTestId='toggle-label'
    >
      {children}
    </Typography>
  );
};
