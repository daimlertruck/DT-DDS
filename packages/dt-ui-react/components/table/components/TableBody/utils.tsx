import { Children, ReactNode, cloneElement, isValidElement } from 'react';

import { TableDataCellHeader } from '../TableDataCell/TableDataCell';

export const addHeadersToDataCells = (
  dataCells: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(dataCells, (dataCell, index) => {
    if (isValidElement(dataCell)) {
      const header = headers[index];
      return cloneElement(
        dataCell,
        {},
        <>
          <TableDataCellHeader>{header}</TableDataCellHeader>
          {dataCell.props.children}
        </>
      );
    }
    return dataCell;
  });
};

export const addHeadersToDataCellsInRows = (
  rows: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(rows, (row) => {
    if (isValidElement(row)) {
      return cloneElement(
        row,
        {},
        addHeadersToDataCells(row.props.children, headers)
      );
    }
    return row;
  });
};
