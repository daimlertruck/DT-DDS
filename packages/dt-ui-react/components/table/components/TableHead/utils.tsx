import { Children, ReactNode, isValidElement } from 'react';

const extractColumnHeaders = (columnHeaders: ReactNode): ReactNode[] => {
  return Children.toArray(columnHeaders).reduce<ReactNode[]>(
    (headers, columnHeader) => {
      if (isValidElement(columnHeader)) {
        return [...headers, columnHeader.props.children];
      }

      return headers;
    },
    []
  );
};

export const extractTableHeaders = (theadRow: ReactNode) => {
  if (isValidElement(theadRow)) {
    return extractColumnHeaders(theadRow.props.children);
  }

  return [];
};
