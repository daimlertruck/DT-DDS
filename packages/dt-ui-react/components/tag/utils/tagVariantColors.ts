import { CustomTheme as Theme } from '../../../types';
import { TagColor, TagVariant } from '../constants';

interface TagVariantColors {
  border: string | number;
  color: string;
  borderColor?: string;
  backgroundColor: string;
  '&:hover'?: {
    borderColor?: string;
    backgroundColor: string;
    color: string;
  };
}

interface TagVariantColorsProps {
  theme: Theme;
  color: TagColor;
  variant: TagVariant;
  hasHover: boolean;
  isDisabled: boolean;
}

const getColorWithShade = ({
  theme,
  color,
  shade,
}: Pick<TagVariantColorsProps, 'color'> & {
  shade: '90' | '60' | '30' | '10';
  theme: Theme;
}) => {
  const { colors, palette } = theme;

  if (color === 'primary') {
    return {
      '90': palette.primaryDark_500,
      '60': palette.primaryDark_200,
      '30': palette.primaryLight_100,
      '10': palette.primaryLight_400,
    }[shade];
  }

  const colorWithShade: keyof Theme['colors'] = `${color}_${shade}`;

  return colors[colorWithShade];
};

export const tagVariantColors = ({
  theme,
  color,
  variant,
  hasHover,
  isDisabled,
}: TagVariantColorsProps): TagVariantColors => {
  const colorShade90 = getColorWithShade({
    theme,
    color,
    shade: '90',
  });
  const colorShade60 = getColorWithShade({
    theme,
    color,
    shade: '60',
  });
  const colorShade30 = getColorWithShade({
    theme,
    color,
    shade: '30',
  });
  const colorShade10 = getColorWithShade({
    theme,
    color,
    shade: '10',
  });

  const colorWithDisabled = isDisabled ? colorShade30 : colorShade60;

  return {
    solid: {
      border: theme.spacing.none,
      color: theme.colors.white,
      backgroundColor: colorWithDisabled,
      ...(hasHover && {
        '&:hover': {
          backgroundColor: colorShade90,
          color: theme.colors.white,
        },
      }),
    },
    outlined: {
      border: `1px solid ${colorShade60}`,
      color: colorWithDisabled,
      backgroundColor: colorShade10,
      borderColor: colorWithDisabled,
      ...(hasHover && {
        '&:hover': {
          borderColor: colorShade90,
          backgroundColor: colorShade10,
          color: colorShade90,
        },
      }),
    },
  }[variant];
};
