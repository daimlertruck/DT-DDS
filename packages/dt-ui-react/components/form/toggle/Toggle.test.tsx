import { render } from '@testing-library/react';
import React from 'react';

import { withProviders } from '../../../utils';

import { Toggle } from './Toggle';

describe('<Toggle /> component', () => {
  const ProvidedToggle = withProviders(Toggle);
  const LABEL = 'Toggle';

  describe('when the toggle is disabled and is checked', () => {
    it('should render a toggle with grey 400 bg color and light grey label color', () => {
      const { container } = render(
        <ProvidedToggle
          isDisabled={true}
          isChecked={true}
          onChange={() => null}
        >
          <Toggle.Label>{LABEL}</Toggle.Label>
        </ProvidedToggle>
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe("when the toggle is disabled and isn't checked", () => {
    it('should render a toggle with 300 grey bg color and light grey label color', () => {
      const { container } = render(
        <ProvidedToggle
          isDisabled={true}
          isChecked={false}
          onChange={() => null}
        >
          <Toggle.Label>{LABEL}</Toggle.Label>
        </ProvidedToggle>
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe("when the toggle isn't disabled and is checked with default state", () => {
    it('should render a toggle with green color and a label', () => {
      const { container } = render(
        <ProvidedToggle
          isDisabled={false}
          isChecked={true}
          onChange={() => null}
        >
          {LABEL}
        </ProvidedToggle>
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe("when the toggle isn't disabled and isn't checked", () => {
    it('should render a toggle with 400 gray and a label', () => {
      const { container } = render(
        <ProvidedToggle
          isDisabled={false}
          isChecked={false}
          onChange={() => null}
        >
          {LABEL}
        </ProvidedToggle>
      );

      expect(container).toMatchSnapshot();
    });
  });
});
