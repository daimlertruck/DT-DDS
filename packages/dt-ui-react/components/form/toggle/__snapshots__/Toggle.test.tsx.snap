// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle /> component when the toggle is disabled and is checked should render a toggle with grey 400 bg color and light grey label color 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  opacity: 0;
  z-index: 1;
  border-radius: 15px;
  width: 40px;
  height: 20px;
  cursor: pointer;
}

.emotion-2:disabled {
  cursor: auto;
}

.emotion-2:checked+label {
  background: #8B8C8C;
}

.emotion-2:checked+label::after {
  content: '';
  display: block;
  border-radius: 50%;
  width: 16px;
  height: 16px;
  margin-left: 22px;
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

.emotion-2:disabled~span {
  color: #8B8C8C;
}

.emotion-2:not(:checked)+label {
  background: #323232;
}

.emotion-4 {
  position: absolute;
  width: 40px;
  height: 20px;
  border-radius: 10px;
  background: #8B8C8C;
  cursor: pointer;
}

.emotion-4::after {
  content: '';
  display: block;
  border-radius: 50%;
  width: 16px;
  height: 16px;
  margin: 2px;
  background: #ffffff;
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

.emotion-6 {
  font-weight: 400;
  font-size: 12px;
  line-height: 16.08px;
  letter-spacing: normal;
  margin: 0;
  color: #4F4F4F;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="toggle"
  >
    <input
      checked=""
      class="emotion-2 emotion-3"
      disabled=""
      id="toggle-checkbox-id"
      type="checkbox"
    />
    <label
      class="emotion-4 emotion-5"
      for="toggle-checkbox-id"
    />
    <span
      class="emotion-6 emotion-7"
      color="gray_200"
      data-testid="toggle-label"
      style="font-weight: 700;"
    >
      Toggle
    </span>
  </div>
</div>
`;

exports[`<Toggle /> component when the toggle is disabled and isn't checked should render a toggle with 300 grey bg color and light grey label color 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  opacity: 0;
  z-index: 1;
  border-radius: 15px;
  width: 40px;
  height: 20px;
  cursor: pointer;
}

.emotion-2:disabled {
  cursor: auto;
}

.emotion-2:checked+label {
  background: #8B8C8C;
}

.emotion-2:checked+label::after {
  content: '';
  display: block;
  border-radius: 50%;
  width: 16px;
  height: 16px;
  margin-left: 22px;
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

.emotion-2:disabled~span {
  color: #8B8C8C;
}

.emotion-2:not(:checked)+label {
  background: #323232;
}

.emotion-4 {
  position: absolute;
  width: 40px;
  height: 20px;
  border-radius: 10px;
  background: #8B8C8C;
  cursor: pointer;
}

.emotion-4::after {
  content: '';
  display: block;
  border-radius: 50%;
  width: 16px;
  height: 16px;
  margin: 2px;
  background: #ffffff;
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

.emotion-6 {
  font-weight: 400;
  font-size: 12px;
  line-height: 16.08px;
  letter-spacing: normal;
  margin: 0;
  color: #4F4F4F;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="toggle"
  >
    <input
      class="emotion-2 emotion-3"
      disabled=""
      id="toggle-checkbox-id"
      type="checkbox"
    />
    <label
      class="emotion-4 emotion-5"
      for="toggle-checkbox-id"
    />
    <span
      class="emotion-6 emotion-7"
      color="gray_200"
      data-testid="toggle-label"
      style="font-weight: 700;"
    >
      Toggle
    </span>
  </div>
</div>
`;

exports[`<Toggle /> component when the toggle isn't disabled and is checked with default state should render a toggle with green color and a label 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  opacity: 0;
  z-index: 1;
  border-radius: 15px;
  width: 40px;
  height: 20px;
  cursor: pointer;
}

.emotion-2:disabled {
  cursor: auto;
}

.emotion-2:checked+label {
  background: #5fa50b;
}

.emotion-2:checked+label::after {
  content: '';
  display: block;
  border-radius: 50%;
  width: 16px;
  height: 16px;
  margin-left: 22px;
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

.emotion-2:disabled~span {
  color: #8B8C8C;
}

.emotion-2:not(:checked)+label {
  background: #8B8C8C;
}

.emotion-4 {
  position: absolute;
  width: 40px;
  height: 20px;
  border-radius: 10px;
  background: #8B8C8C;
  cursor: pointer;
}

.emotion-4::after {
  content: '';
  display: block;
  border-radius: 50%;
  width: 16px;
  height: 16px;
  margin: 2px;
  background: #ffffff;
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="toggle"
  >
    <input
      checked=""
      class="emotion-2 emotion-3"
      id="toggle-checkbox-id"
      type="checkbox"
    />
    <label
      class="emotion-4 emotion-5"
      for="toggle-checkbox-id"
    />
    Toggle
  </div>
</div>
`;

exports[`<Toggle /> component when the toggle isn't disabled and isn't checked should render a toggle with 400 gray and a label 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  opacity: 0;
  z-index: 1;
  border-radius: 15px;
  width: 40px;
  height: 20px;
  cursor: pointer;
}

.emotion-2:disabled {
  cursor: auto;
}

.emotion-2:checked+label {
  background: #5fa50b;
}

.emotion-2:checked+label::after {
  content: '';
  display: block;
  border-radius: 50%;
  width: 16px;
  height: 16px;
  margin-left: 22px;
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

.emotion-2:disabled~span {
  color: #8B8C8C;
}

.emotion-2:not(:checked)+label {
  background: #8B8C8C;
}

.emotion-4 {
  position: absolute;
  width: 40px;
  height: 20px;
  border-radius: 10px;
  background: #8B8C8C;
  cursor: pointer;
}

.emotion-4::after {
  content: '';
  display: block;
  border-radius: 50%;
  width: 16px;
  height: 16px;
  margin: 2px;
  background: #ffffff;
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="toggle"
  >
    <input
      class="emotion-2 emotion-3"
      id="toggle-checkbox-id"
      type="checkbox"
    />
    <label
      class="emotion-4 emotion-5"
      for="toggle-checkbox-id"
    />
    Toggle
  </div>
</div>
`;
